Framework/Algorithms/inc/MantidAlgorithms/BinaryOperation.h

```diff
@@ -111,8 +111,8 @@ class DLLExport BinaryOperation : public API::Algorithm {
   /// masked. If so then the output spectra has zeroed data
   /// and is also masked. The function returns true if further processing is not
   /// required on the spectra.
-  virtual bool propagateSpectraMask(const API::MatrixWorkspace_const_sptr lhs,
-                                    const API::MatrixWorkspace_const_sptr rhs,
+  virtual bool propagateSpectraMask(const API::SpectrumInfo &lhsSpectrumInfo,
```

Comment:
Seems sensible. I expect that this function was doing nothing with the shared pointer, and the copy of the function arguments was clearly going to be expensive, as well as being ambiguous (regarding the shared_ptr aspect that is).